app/src/main/java/com/xmartlabs/taskloans/ui/screens/dashboard/tabs/HistoryFragment.kt

```diff
@@ -1,10 +1,87 @@
 package com.xmartlabs.taskloans.ui.screens.dashboard.tabs
 
+import android.os.Bundle
+import android.view.View
+import android.widget.Toast
+import androidx.lifecycle.lifecycleScope
+import androidx.paging.PagingData
+import androidx.paging.flatMap
+import androidx.recyclerview.widget.DividerItemDecoration
+import com.xmartlabs.swissknife.core.extensions.disable
+import com.xmartlabs.swissknife.core.extensions.getDrawableCompat
+import com.xmartlabs.swissknife.core.extensions.gone
+import com.xmartlabs.swissknife.core.extensions.visible
+import com.xmartlabs.taskloans.R
+import com.xmartlabs.taskloans.data.model.EntryUI
+import com.xmartlabs.taskloans.data.model.UserEntry
 import com.xmartlabs.taskloans.databinding.FragmentHistoryBinding
 import com.xmartlabs.taskloans.ui.common.BaseViewBindingFragment
+import com.xmartlabs.taskloans.ui.screens.dashboard.DashboardFragmentViewModel
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.launch
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class HistoryFragment : BaseViewBindingFragment<FragmentHistoryBinding>() {
+  private val viewModel: DashboardFragmentViewModel by viewModel()
+  private var adapter: HistoryAdapter? = HistoryAdapter()
 
   override fun inflateViewBinding(): FragmentHistoryBinding =
       FragmentHistoryBinding.inflate(layoutInflater)
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    setUpRecyclerView()
+    loadHistoryList()
+  }
+
+  override fun onDetach() {
+    super.onDetach()
+    adapter = null
```

Comment:
this is not what should be null at the end the `historyRecyclerView.adapter` is what causes the memory to leek